src/Strategy/IfElse.php

```diff
@@ -4,38 +4,35 @@
 use ScriptFUSION\Mapper\Mapping;
 
 /**
- * Delegates to one expression or another depending on whether the specified condition loosely evaluates to true.
+ * Delegates to one expression or another depending on whether the specified condition strictly evaluates to true.
  */
 class IfElse extends Delegate
 {
-    /** @var Strategy|Mapping|array|mixed */
-    private $if;
+    /** @var callable */
+    private $condition;
 
     /** @var Strategy|Mapping|array|mixed */
     private $else;
 
     /**
      * Initializes this instance with the specified condition, the specified
-     * strategy or mapping to be resolved when condition is non-null and,
-     * optionally, the specified strategy or mapping to be resolved when
-     * condition is null.
+     * expression to be resolved when condition is true and, optionally, the
+     * specified expression to be resolved when condition is false.
      *
-     * @param Strategy|Mapping|array|mixed $condition Condition.
+     * @param callable $condition Condition.
      * @param Strategy|Mapping|array|mixed $if Primary expression.
      * @param Strategy|Mapping|array|mixed|null $else Optional. Fallback expression.
      */
-    public function __construct($condition, $if, $else = null)
+    public function __construct(callable $condition, $if, $else = null)
     {
-        parent::__construct($condition);
-
-        $this->if = $if;
+        $this->condition = $condition;
+        parent::__construct($if);
```

Comment:
Although not enforced, house style is to call the parent first and do assignments after, as before.